src/components/fx/hover.js

```diff
@@ -950,6 +963,21 @@ function createHoverText(hoverData, opts, gd) {
             text = name;
         }
 
+        // hovertemplate
+        var trace = d.trace;
+        var hovertemplate = d.hovertemplate || false;
+        var hovertemplateLabels = d.hovertemplateLabels || d;
+        var eventData = d.eventData[0] || {};
+        if(hovertemplate) {
+            text = Lib.hovertemplateString(hovertemplate, hovertemplateLabels, eventData, trace);
+
+            var EXTRA_STRING_REGEX = /<extra>([\s\S]*)<\/extra>/;
```

Comment:
:cow2: Most browsers probably have a way to optimise this, but we usually define regexes in the outer module scope so we only have to compile them once on script load.

Reply:
@etpinard I was also thinking I could move this whole block into a function of its own.

Reply:
Fixed in 0659d20